### ISSUE TITLE
Add tests for an inline editor on a <p> or <h1>

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-editor-inline#5]
AFAICS https://github.com/ckeditor/ckeditor5-editor-inline/issues/3 discusses how the editor should work on various elements.

Here, I'd just to have a sample for what happens now when you enable an editor on e.g. `<p>`. I can already tell that there may be problems with what kind of root is created in the model etc. So, let's see first some manual test and it will give some ideas what to test automatically.

**Comment by Reinmar** [comment]
Interesting aspect of this is that the editor class needs to know to what model element the DOM element on which the editor is create maps. 

E.g. if you create the editor on a `<p>` the root model element used by the editor should be of a `$block` type. Without the paragraph feature the editor won't even know that `<p>` must be converted to `<paragraph>` and that `paragraph` extends that `$block` type. However, I don't think that it should actually work through conversion here because we're not converting roots.

It's tricky... :)

**Comment by oleq** [comment]
Can't wait to see the demo. It's a killer feature for complex forms and content editing systems.

**Comment by VincentClair** [comment]
Hi, do you have some news on ck5 supporting other elements (Hn, P, etc.) than DIV ?
I tried 0.11 without success.

**Comment by Reinmar** [comment]
We didn't work on this yet. We'll check soon if we can give it a try, but I can't promise anything yet (also because we don't know how much work it will require).

**Comment by msamsel** [comment]
@Reinmar I create a PR which adds manual test for inline editor. Test contains editor created over `<h1>`, `<p>`, `<blockqoute>` and `<pre>` tags.
Does is solves case from this task?

**Comment by msamsel** [comment]
@Reinmar more detailed analysis related to this topic.

I add branch [`t/ckeditor5-editor-inline/5`](https://github.com/ckeditor/ckeditor5/tree/t/ckeditor5-editor-inline/5) to ckeditor5 repository, which applies the appropriate modification to the engine and the editor-inline repository. However, there is very little to modify. There is registered a new element in schema `$inlineRoot`, modification in `$text` to be allowed in `$inlineRoot` and change in `InlineEditor` constructor to use based on a configuration option.

There is a manual test `editor-inline/tests/manual/inlineroot`, which is POC how this works. I've tested it with some of the editor's features and it seems to work really well.
* I haven't spotted any errors during editor usage in this configuration.
* All block related features are automatically disabled.
* Inline features like fonts, basic-styles, highlight, links, etc. are working.
* Undo and redo is available and works nice.
* Clipboard paste only inline content. It extracts content even with inline styles and pastes it without blocks.

I noticed one thing, which is rather related to missing feature in the editor. Inline images are removed during upcast. As images in editor became block element and that's why those are lost during conversion.

![Screenshot 2019-07-09 at 10 03 55](https://user-images.githubusercontent.com/9017322/60871278-b7f71a80-a232-11e9-8ea8-55d9c76cb604.png)
